src/thread/mod.rs

```diff
@@ -25,6 +26,8 @@ pub enum VMError {
     ExpectedVariableStack(bool),
     #[error(transparent)]
     BadType(#[from] TypeError),
+    #[error(transparent)]
+    BadCall(#[from] MetaCallError),
```

Comment:
This is not your fault at all, but the error situation with `run_vm` is a mess and this makes it more apparent.

I think with this change, `MetaCallError` would **sometimes** be turned into a `VMError` and sometimes not. I believe you added this because `LuaFrame::call_function` needs to return `VMError`, right? I think the proper change is to make `run_vm` return `VMError` rather than the currently super generic `RuntimeError` (and add variants to `VMError` as appropriate), but this also means that some of the error types could be redesigned (for example, it's maybe not necessary to have a separate `MetaCallError` type then?)

Also, I believe the *entire* reason that `run_vm` currently needs to return `RuntimeError` is because of all the operators that SHOULD have been meta ops but weren't.

Reply:
Besides for the `MetaCallError` issues, there are ~3 remaining cases in `run_vm` that don't fit with the current error types:
- Numeric for loops are implemented entirely within `run_vm`, and have two error cases (currently just using `BinaryOperatorError`s as a placeholders); those seem like they should be turned into explicit VMError variants (with accurate names/data)
(Also, this seems like it could overflow:)
https://github.com/kyren/piccolo/blob/091a9565e5e09a68b28a1d7b164c7edb75aaf8c3/src/thread/vm.rs#L308 
-  `Less` and `LessEq` are not yet implemented as meta operators, but will be in the future.  Replacing their error types with the `MetaOperatorError` equivalent is easy.
- `String::concat` should probably also be turned into a metamethod?  It'll be significantly different from the others, though, since it's variadic and will need to use a sequence internally.  For now, adding a case for it in `VMError` should work, but it should eventually end up as a `MetaOperatorError`